### PR TITLE
Use Cassandra 4.0.0 GA

### PR DESCRIPTION
**Pull request by dougwettlaufer**

### CODE REVIEW

File: persistence-cassandra-4.0/pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
   <artifactId>persistence-cassandra-4.0</artifactId>
   <properties>
     <!-- If you update this, make sure to keep `cassandra.bundled-driver.version` in sync -->
-    <cassandra.version>4.0-rc2</cassandra.version>
+    <cassandra.version>4.0</cassandra.version>
```
Comment (mpenick) [inline persistence-cassandra-4.0/pom.xml:13]: Does this need to be `4.0.0`?

Comment (dougwettlaufer) [reply persistence-cassandra-4.0/pom.xml]: Yep

**Review by mpenick — APPROVED** [review]

**Review by prabhatjha — APPROVED** [review]
Finally. :)

**Review by dimas-b — APPROVED** [review]
Lest we forget, we still have `4.0-alpha4` in `persistence-api`. I'm sure it is not critical to switch to 4.0.0 in Persistence, but I guess it might be worth updating that at some point too. WDYT?

**Comment by dimas-b** [comment]
> Step #4 - "cassandra-4.0": ERROR [Exec Stream Pumper] 2021-07-26 21:30:48,691 CcmBridge.java:374 - ccmerr> 21:30:48,690 ccm WARNING Downloading 4.0.0 failed, trying to build from git instead.
> Step #4 - "cassandra-4.0": ERROR [Exec Stream Pumper] 2021-07-26 21:30:48,691 CcmBridge.java:374 - ccmerr> The error was: Invalid version 4.0.0 (underlying error is: HTTP Error 404: Not Found)

@dougwettlaufer : Do we need to rebuild our CI docker image to get the latest `ccm` compatible with C* 4.0.0? It appears to work fine for me locally :thinking:

**Comment by dougwettlaufer** [comment]
> Lest we forget, we still have `4.0-alpha4` in `persistence-api`. I'm sure it is not critical to switch to 4.0.0 in Persistence, but I guess it might be worth updating that at some point too. WDYT?

Good catch. Updating

**Comment by ivansenic** [comment]
So DSE build fails and my analysis shows the following:

* the error started with the commit https://github.com/stargate/stargate/commit/1ca3e41cec16ad68d8d164c2da806dd16203d531 that updated the builder to the `1.0.5`
* previously the builder was on `1.0.3`, however this image was built on Nov 12, 2020, meaning that there were bunch of the commits altering the Dockerfile (https://github.com/stargate/stargate/commits/1ca3e41cec16ad68d8d164c2da806dd16203d531/ci/Dockerfile)

In order to confirm this I am gonna move the builder image back to the `1.0.3` in the `cloudbuild-dse`..

**Comment by ivansenic** [comment]
OK I am gonna merge this as this is best we can do at the moment. The DSE build still runs with the stargate-builder `v1.0.3`, but this was anyway the case until now as well. The issue #1142 should tackle that.

**Pull request merged**